specs/Deployment/DeploymentAPI.md

```diff
@@ -0,0 +1,141 @@
+should be used to validate the WinAppSDK is the correct version and all required packages are present.
+
+The developer scenarios for this API is to answer the questions:
+1. Does this system have the min version of the Windows App SDK I require and are they usable?
+2. What do I do to resolve this if it doesn't?
+
+The API addresses #1 by a simple query of the version they seek and a possible check for OS
+updates. #2 has two possible answers. First is direct resolution by the developer by acquiring and
+installing the packages if possible. Since the main packages are small and the target developer
+scenario will have permissions to deploy them, we will minimize developer and user pain by
+including the main package MSIX packages within the framework, and this API will allow for the
+deployment of that package if needed.
+
+Processes lacking permission to deploy MSIX are out-of-scope at this time.
+
+# Examples
+
+## GetStatus
+
+This is the means by which an app or program can call to verify all required packages are present
+to a minimum version needed by the loaded framework.  A process should call GetStatus() after
+process initialization and before using Windows App Runtime content (e.g. in main() or WinMain()).
+
+Developers call this method without parameters, as all information about the version, channel, and
+architecture are derived from the current WARfwk loaded. This method is good to identify if there
+is possible work required in the form of package install which may take several seconds to
+complete.
+
+```C++
+auto status{Microsoft::Windows::ApplicationModel::WindowsAppRuntime::Deployment::GetStatus()};
+if (!status.IsOK())
+{
+    ...
+}
+```
+
+## Initialize
+
+This API performs a status check and if any packages are required for install, attempts to register
+those packages and get the WindowsAppRuntime into an OK state. This will perform package install if
+possible. It is possible multiple packages may be required for install, and this will attempt
+to deploy all required packages. The returned WindowsAppRuntimeStatus will contain the first error
+if one occurs during package install. There will be no rollback of successfully installed packages
+if an error occurs on a subsequent package install.
+
+
+```C++
+auto status{Microsoft::Windows::ApplicationModel::WindowsAppRuntime::Deployment::Initialize()};
+if (!status.IsOK())
+{
+    return status.ExtendedError();
+}
+```
+
+
+# API Details
+
+```c#
+namespace Microsoft.Windows.ApplicationModel.WindowsAppSDK
+{
+    /// Represents the current Deployment status of the WindowsAppRuntime
+    [experimental]
+    runtimeclass DeploymentStatus
+    {
+        DeploymentStatus(
+            Boolean isOK,
+            Boolean packageInstallRequired,
+            HRESULT extendedError);
+
+        /// Returns True if WindowsAppRuntime has all required packages at equal or greater version
```

Comment:
When can both booleans be `true`? Or `false`?  In most WinRT "alternate success" models there's a `Result` type that encodes an enum plus an extended error code.  Like:

```c#
enum WinAppSDKDeploymentStatus {
    Ok = 0,
    PackageInstallRequired,
    PackageInstallFailed,
    Other,
}
runtimeclass WinAppSDKDeploymentResult
{
    WinAppSDKDeploymentResult(WinAppSDKDeploymentStatus status, HRESULT extendedError);
    WinAppSDKDeploymentStatus Status { get; };
    HRESULT ExtendedError { get; }
}
```

Reply:
Agree, I will change to use something similar to this.